Source/Infrastructure/Kysect.Shreks.Seeding/EntityGenerators/User/StudentGenerator.cs

```diff
@@ -26,18 +26,16 @@ protected override Student Generate(int index)
     {
         if (index >= _userGenerator.GeneratedEntities.Count)
             throw new IndexOutOfRangeException("Student count is greater than count of users.");
-        
-        var groupCount = _studentGroupGenerator.GeneratedEntities.Count;
-        var groupNumber = index % groupCount;
 
-        StudentGroup group = _studentGroupGenerator.GeneratedEntities[groupNumber];
+        User user = _userGenerator.GeneratedEntities[index];
 
-        var user = _userGenerator.GeneratedEntities[index];
+        StudentGroup[] groups = _studentGroupGenerator.GeneratedEntities
+            .Where(x => x.Students.Any(student => student.User.Equals(user)) is false)
```

Comment:
🤔  Интересно, а можно не тянуть всех юзеров на старте приложения?
```suggestion
            .Except(student.User)
```

Reply:
`IEnumerable<StudyGroup>` Except `Student`?

Reply:
Это же сидер, эти данные изначально в памяти лежат, а потом в базу кладутся